File: python/majority-element.py
from typing import List
def majorityElement(nums: List[int]) -> int:
    el_count = {}
    for num in nums:
        if num in el_count:
            el_count[num] = el_count[num] + 1
        else:
            el_count[num] = 1
    return max(el_count, key=el_count.get)

if __name__ == "__main__":
    print(majorityElement([3,2,3]))

File: rust/src/majority_element.rs
use std::collections::HashMap;

fn majority_element(nums: Vec<i32>) -> i32 {
    let mut count_map: HashMap<i32, i32> = HashMap::new();
    for num in nums {
        *count_map.entry(num).or_insert(0) += 1;
    }

    let (key_of_max_value, _max_value) = count_map
        .iter()
        .max_by(|a, b| a.1.cmp(&b.1))
        .unwrap_or((&0, &0));

    return *key_of_max_value;
}
